sql-cli/sql_cli/connections.py

```diff
@@ -0,0 +1,99 @@
+from airflow.models import Connection
+from airflow.utils.session import create_session
+from marshmallow.exceptions import ValidationError
+
+from sql_cli.settings import SQL_CLI_PROJECT_DIRECTORY
+
+
+def _load_yaml_connections(environment: str) -> list[dict[str, Any]]:
+    """Gets the configuration yaml for the given environment and loads the connections from it into a dictionary"""
+    config_file = (
+        SQL_CLI_PROJECT_DIRECTORY / "config" / environment / "configuration.yaml"
+    )
+    if not config_file.exists():
+        raise FileNotFoundError(
+            f"Config file configuration.yaml does not exist for environment {environment}"
+        )
+
+    connections = yaml.safe_load(open(config_file))["connections"]
+    return connections
+
+
+def _test_connection(conn_obj: Connection, environment: str) -> bool:
+    """Tests whether connection is established successfully with the given data."""
+    conn_id = conn_obj.conn_id
+    try:
+        status, message = conn_obj.test_connection()
+        if not status:
+            logging.info(
+                "Connection %s failed in %s environment with error %s",
+                conn_id,
+                environment,
+                message,
+            )
+            return False
+        logging.info(
+            "Connection %s tested successfully in %s environment", conn_id, environment
+        )
+        return True
+    except ValidationError as err:
+        raise BadRequest(detail=str(err.messages))
+
+
+def _create_or_replace_connection(conn_obj: Connection, environment: str) -> None:
+    """Creates a new or replaces existing connection in the Airflow DB with the given connection object."""
+    conn_id = conn_obj.conn_id
+    with create_session() as session:
+        db_connection = (
+            session.query(Connection).filter_by(conn_id=conn_id).one_or_none()
+        )
+        if db_connection:
+            session.delete(db_connection)
+            session.commit()
+            logging.info(
+                "Existing connection %s deleted successfully in %s environment",
+                conn_id,
+                environment,
+            )
+        session.add(conn_obj)
+        session.commit()
+        logging.info(
+            "Connection %s added successfully in %s environment", conn_id, environment
+        )
+
+
+def validate_connections(
+    environment: str = "default", connection: str | None = None
+) -> None:
+    """
+    Validates that the given connections are valid and registers them to Airflow with replace policy for existing
+    connections.
+    """
+    config_file_contains_connection = False
+    connections = _load_yaml_connections(environment)
+    for conn in connections:
+        conn_id = conn["conn_id"]
+        conn["connection_id"] = conn_id
+        conn.pop("conn_id")
+        data = connection_schema.load(conn)
+        if connection and conn_id != connection:
```

Comment:
I think `connection` is the wrong word here. Wouldn't `connection_id`/`conn_id` make more sense? It begins already in the `validate` app command.

Reply:
I referred for this name for this exposed flag in the validate command from the PRD, where it says connection: https://docs.google.com/document/d/1_iUp0Mci3ZTdcRUUHTvx8F_lffxUv601uBkpJBMhiXs/edit#

Reply:
@mikeshwe Is it okay if we call it `connection_id` instead of `connection` as the exposed flag for the `validate` command?

Reply:
Renamed it to connection_id for internal functions.